components/server/src/server.ts

```diff
@@ -308,8 +309,19 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
         const gauge = new prom.Gauge({
             name: `server_websocket_connection_count`,
             help: 'Currently served websocket connections',
+            labelNames: ["clientType"],
         });
-        this.websocketConnectionHandler.onConnectionCreated(() => gauge.inc());
-        this.websocketConnectionHandler.onConnectionClosed(() => gauge.dec());
+        this.websocketConnectionHandler.onConnectionCreated((_, req) => gauge.inc({ clientType: WebsocketClientType.getClientType(req) || "undefined" }));
+        this.websocketConnectionHandler.onConnectionClosed((_, req) => gauge.dec({ clientType: WebsocketClientType.getClientType(req) || "undefined" }));
+    }
+
+    protected installWebsocketClientContextGauge() {
+        const gauge = new prom.Gauge({
+            name: `server_websocket_client_context_count`,
+            help: 'Currently served client contexts',
+            labelNames: ["authLevel"],
+        });
+        this.websocketConnectionHandler.onClientContextCreated((ctx) => gauge.inc({ authLevel: ctx.authLevel }));
```

Comment:
I see, so that's the point. We're registering analytics here. 
I'm curious how these events could be used further for managing/reusing connections or parts of them.